packages/core/src/State.ts

```diff
@@ -90,111 +92,75 @@ export class State<
     TEvent extends EventObject = EventObject
   >(
     stateValue: State<TContext, TEvent, any> | StateValue,
-    context: TContext = {} as TContext
+    context: TContext = {} as TContext,
+    machine: AnyStateMachine
   ): State<TContext, TEvent, any> {
     if (stateValue instanceof State) {
       if (stateValue.context !== context) {
-        return new State<TContext, TEvent>({
-          value: stateValue.value,
-          context,
-          _event: stateValue._event,
-          _sessionid: null,
-          actions: [],
-          meta: {},
-          configuration: [], // TODO: fix,
-          transitions: [],
-          children: {}
-        });
+        return new State<TContext, TEvent>(
+          {
+            value: stateValue.value,
+            context,
+            _event: stateValue._event,
+            _sessionid: undefined,
+            actions: [],
+            meta: {},
+            configuration: [], // TODO: fix,
```

Comment:
I know this is not changed in this PR, but I'm just curious as to what the `TODO: fix` comment refers to?

Reply:
The configuration array should never be empty. I can actually fix this in this PR since the `State` should only ever be created from a machine.

Reply:
That makes sense; thank you for the explanation. And if you can fix it in the scope of this PR, that would be ideal.